moveit_core/collision_detection_bullet/test/test_bt_collision_detection.cpp

```diff
@@ -0,0 +1,558 @@
+  // this is not really a failure; it is just that slow;
+  // looking into doing collision checking with a voxel grid.
+  ROS_INFO_NAMED("collision_detection.bullet", "Adding boxes took %g", t);
+}
+
+TEST_F(BulletCollisionDetectionTester, MoveMesh)
+{
+  robot_state::RobotState robot_state1(robot_model_);
+  robot_state1.setToDefaultValues();
+  robot_state1.update();
+
+  Eigen::Isometry3d kinect_pose;
+  kinect_pose.setIdentity();
+  shapes::ShapePtr kinect_shape;
+  kinect_shape.reset(shapes::createMeshFromResource(kinect_dae_resource_));
+
+  cworld_->getWorld()->addToObject("kinect", kinect_shape, kinect_pose);
+
+  Eigen::Isometry3d np;
+  for (unsigned int i = 0; i < 5; i++)
+  {
+    np = Eigen::Translation3d(i * .001, i * .001, i * .001) * Eigen::Quaterniond::Identity();
+    cworld_->getWorld()->moveShapeInObject("kinect", kinect_shape, np);
+    collision_detection::CollisionRequest req;
+    collision_detection::CollisionResult res;
+    cworld_->checkCollision(req, res, *crobot_, robot_state1, *acm_);
+  }
+}
+
+TEST_F(BulletCollisionDetectionTester, TestChangingShapeSize)
+{
+  robot_state::RobotState robot_state1(robot_model_);
+  robot_state1.setToDefaultValues();
+  robot_state1.update();
+
+  collision_detection::CollisionRequest req1;
+  collision_detection::CollisionResult res1;
+
+  ASSERT_FALSE(res1.collision);
+
+  EigenSTL::vector_Isometry3d poses;
+  std::vector<shapes::ShapeConstPtr> shapes;
+  for (unsigned int i = 0; i < 5; i++)
+  {
+    cworld_->getWorld()->removeObject("shape");
+    shapes.clear();
+    poses.clear();
+    shapes.push_back(shapes::ShapeConstPtr(new shapes::Box(1 + i * .0001, 1 + i * .0001, 1 + i * .0001)));
+    poses.push_back(Eigen::Isometry3d::Identity());
+    cworld_->getWorld()->addToObject("shape", shapes, poses);
+    collision_detection::CollisionRequest req;
+    collision_detection::CollisionResult res;
+    cworld_->checkCollision(req, res, *crobot_, robot_state1, *acm_);
+    ASSERT_TRUE(res.collision);
+  }
+
+  Eigen::Isometry3d kinect_pose = Eigen::Isometry3d::Identity();
+  shapes::ShapePtr kinect_shape;
+  kinect_shape.reset(shapes::createMeshFromResource(kinect_dae_resource_));
+  cworld_->getWorld()->addToObject("kinect", kinect_shape, kinect_pose);
+  collision_detection::CollisionRequest req2;
+  collision_detection::CollisionResult res2;
+  cworld_->checkCollision(req2, res2, *crobot_, robot_state1, *acm_);
+  ASSERT_TRUE(res2.collision);
+  for (unsigned int i = 0; i < 5; i++)
+  {
+    cworld_->getWorld()->removeObject("shape");
+    shapes.clear();
+    poses.clear();
+    shapes.push_back(shapes::ShapeConstPtr(new shapes::Box(1 + i * .0001, 1 + i * .0001, 1 + i * .0001)));
+    poses.push_back(Eigen::Isometry3d::Identity());
+    cworld_->getWorld()->addToObject("shape", shapes, poses);
+    collision_detection::CollisionRequest req;
+    collision_detection::CollisionResult res;
+    cworld_->checkCollision(req, res, *crobot_, robot_state1, *acm_);
```

Comment:
This is an exact copy of [this FCL test](https://github.com/ros-planning/moveit/blob/master/moveit_core/collision_detection_fcl/test/test_fcl_collision_detection.cpp) which I use to see if the Bullet functionality matches FCL. I think it would make sense to have the same test scenario (and only a single file) for both of them somewhere (neither in the bullet folder, nor in the fcl folder).

Reply:
I agree in principle, Gtest has some pretty neat tricks for running exact same tests on multiple classes. But I don't know of a good place to put that without breaking the modularity of the packages. Maybe someone else has some good ideas.

Reply:
But both are part of `moveit_core` so this shouldn't be a problem, right?

Reply:
Yeah, but each of the `moveit_core` subfolders kinda act like their own packages. 

Not sure how GTest [typed tests](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#typed-tests) handle being included from other packages, but we could define the base checks in `collision_detection`, and then just include and instantiate the test from each FCL and Bullet folder with 

```
using MyTypes = ::testing::Types<CollisionAllocatorFCL>; // or Bt  
TYPED_TEST_SUITE(FooTest, MyTypes);
```

Reply:
I opened a separate PR for this #1543 and kept the test in this PR as it is. As soon as it is merged, I rebase.